### PR TITLE
Enable variables visible in test case

### PR DESCRIPTION
**Pull request by squirrelsc**
If a variable has is_case_visible flag, it will be passed into test
cases.

### CODE REVIEW

File: lisa/tests/test_testsuite.py [hunk 1]
```diff
@@ -227,39 +227,59 @@ def test_test_result_canrun(self) -> None:
...
             self.assertEqual("before_suite: failed", result.message)
 
     def test_pass_after_suite_failed(self) -> None:
         test_suite = self.generate_suite_instance()
         test_suite.set_fail_phase(fail_on_after_suite=True)
-        test_suite.start(environment=self.default_env, case_results=self.case_results)
+        test_suite.start(
+            environment=self.default_env,
+            case_results=self.case_results,
+            case_variables={},
```
Comment (glima) [inline lisa/tests/test_testsuite.py:256]: Can we get a (testing) test assert that it gets some of these?

**Comment by glima** [comment]
Other than that LGTM

File: lisa/tests/test_testsuite.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/tests/test_testsuite.py:256]: added

File: lisa/runner.py [hunk 2]
```diff
@@ -167,7 +179,10 @@ def _generate_runners(self, runbook: schema.Runbook) -> Iterator[BaseRunner]:
             # keep filters to current runner's only.
             runbook.testcase = parse_testcase_filters(raw_filters)
             runner = factory.create_by_type_name(
-                type_name=runner_name, runbook=runbook, index=len(self._runners)
+                type_name=runner_name,
+                runbook=runbook,
+                index=len(self._runners),
+                case_variables=case_variables,
```
Comment (glima) [inline lisa/runner.py:185]: Might as well inline that call here

**Review by glima — APPROVED** [review]

File: lisa/runner.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/runner.py:185]: If so, that would be called multiple times in the loop.

Comment (glima) [reply lisa/runner.py:185]: Oops, disregard. /me grabs more coffee

**Pull request merged**